### DISCUSSION TITLE
KL loss scaled

### DISCUSSION THREAD

**Discussion by yjgeno** [pyg-team/pytorch_geometric#4483] [Q&A] [answered]
Hi, just wanted to make sure whether the kl loss has to be scaled by the # of nodes, as many tutorials indicate: 
`model.kl_loss() / data.num_nodes`

This is the source defining kl_loss:
https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/models/autoencoder.py#L180
Is the output a mean value of kl_loss?

**Comment by rusty1s** [comment, marked as answer]
Yes, it has be to scaled, see [here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/autoencoder.py#L93). Please also see discussions around this in https://github.com/tkipf/gae/issues/20.

**Reply by yjgeno** [reply]
Thanks this is insightful.
